From f8f7a5ae6e6e4c337ea0bbe0e1869063931dd8df Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 04 Jul 2018 15:50:36 +0300
Subject: [PATCH] board: habana: improve remote power script

Adjust remote power script to work with both, t32/vivato and pdu.
Currently one needs to choose between the two by commenting and
uncommenting multiple lines in the script.
Narrow the commenting game to one line in the script beginning.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/remote_power b/board/habanalabs/common/remote_power
index ffddbd9..8900eeb 100755
--- a/board/habanalabs/common/remote_power
+++ b/board/habanalabs/common/remote_power
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+power=t32_viv
+#power=pdu
+
 cd `dirname $0`
 
 if [ $# -eq 2 ]; then
@@ -25,14 +28,31 @@
 	;;
 esac
 
+case "$power" in
+t32_viv)
+	state="./t32/t32.sh -l"
+	on="./t32/t32.sh full"
+	off="./vivado_lab/viv.sh $board erase"
+	;;
+pdu)
+	state="./ip_pdu $board -l"
+	on="./ip_pdu $board on"
+	off="./ip_pdu $board off"
+	;;
+*)
+	echo "error - Unsupported power protocol: $power"
+	exit 5
+	;;
+esac
+
 case "$action" in
 -l)
-	status=`./t32/t32.sh -l`
-	case "$status" in
-	FALSE)
+	status=`$state`
+	case "${status,,}" in
+	FALSE|off)
 		echo off
 		;;
-	TRUE)
+	TRUE|on)
 		echo ON
 		;;
 	*)
@@ -42,15 +62,10 @@
 	esac
 	;;
 on)
-	./t32/t32.sh full
-	#./ip_pdu $board off
-	#./ip_pdu $board on
+	$on
 	;;
 off)
-	#./ip_pdu $board on
-	#sleep 180
-	./vivado_lab/viv.sh $board erase
-	#./ip_pdu $board off
+	$off
 	;;
 *)
 	echo "error - Unknown action: $action"
